### ISSUE TITLE
kingfisher-process: Changes to postgresql.conf are not in Salt

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/deploy#144]
An easy way to see non-default values:

```sql
SELECT name, current_setting(name), SOURCE
  FROM pg_settings
  WHERE SOURCE NOT IN ('default', 'override');
```

https://wiki.postgresql.org/wiki/Server_Configuration

**Comment by jpmckinney** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Kingfisher: Changes to postgresql.conf are not in Salt](https://trello.com/c/ssC6B4tA/222-kingfisher-changes-to-postgresqlconf-are-not-in-salt)

**Comment by jpmckinney** [comment]
Once this is done we can create a new issue to tune PostgreSQL: https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server

I see `shared_buffers` is already changed, for instance.

**Comment by odscjames** [comment]
I installed an Ubuntu 18 LTS and postgres-11 from the same package. I then compared /etc/postgresql/11/main. pg_hba.conf is already in salt. As expected, only other file with changes is postgresql.conf. Diff is:

[postgresql.conf.txt](https://github.com/open-contracting/deploy/files/4763945/postgresql.conf.txt)

It's a messy diff, which mainly seems to be down to the pg-10 to pg-11 upgrade. Next step is to work out what changes are actually important and compare that to the in server check you suggest. 

For ref, here is the original postgresql.conf as a text file so github is happy: 
[postgresql.conf.original.txt](https://github.com/open-contracting/deploy/files/4763972/postgresql.conf.original.txt)


ps. I've read before uploading and I think there is no security issues in making public but if you disagree delete this comment

**Comment by odscjames** [comment]
[newpostgres.txt](https://github.com/open-contracting/deploy/files/4764062/newpostgres.txt)

Here's what the diff gives. It's untested. I haven't had time to check against in server yet. 

But so far I'd question if we still need WAL, and I'd see if I can update 10->11 inthe 2 old settings without causing issues, just to avoid confusion later.

**Comment by odscjames** [comment]
I have now checked the settings in the database. Everything listed there is either set by the client, the diff file I posted above or the original distribution config so that is fine.

I guess for procedure would be to put the diff file in  /etc/postgresql/11/main/conf.d/kingfisher.conf and have salt update that. When we deploy we could put the postgresql.conf file back to the original distribution by hand. (I did a basic test in VM)

I will now leave this for a few days to give others the chance to comment.

**Comment by jpmckinney** [comment]
All sounds reasonable. It will be easier to review as a PR, which I'd want Dogsbody to review, which might include making some changes to settings.

**Comment by odscjames** [comment]
https://github.com/open-contracting/deploy/pull/162

**Comment by Bjwebb** [comment]
PR is merged now.
